src/Microsoft.VisualStudio.Services.Agent/Util/IOUtil.cs

```diff
@@ -85,5 +86,82 @@ public static string GetTasksPath(IHostContext hostContext)
                 Constants.Path.TasksDirectory);
         }
+
+        public static string Which(String command)
+        {
+            ArgUtil.NotNullOrEmpty(command, nameof(command));
+
+            string path = Environment.GetEnvironmentVariable("Path");
+            if (string.IsNullOrEmpty(path))
+            {
+                return null;
+            }
+
+#if OS_WINDOWS
+            char pathSep = ';';
+#else
+            char pathSep=':';
+#endif
+
+            string[] pathSegments = path.Split(new Char[] { pathSep }, StringSplitOptions.RemoveEmptyEntries);
+            for (int i = 0; i < pathSegments.Length; i++)
+            {
+                pathSegments[i] = Environment.ExpandEnvironmentVariables(pathSegments[i]);
+            }
+
+            foreach (string pathSegment in pathSegments)
+            {
+                if (!string.IsNullOrEmpty(pathSegment) && Directory.Exists(pathSegment))
+                {
+                    string[] matches;
+#if OS_WINDOWS
+                    string pathExt = Environment.GetEnvironmentVariable("PATHEXT");
+                    if (string.IsNullOrEmpty(pathExt))
+                    {
+                        // XP's system default value for PATHEXT system variable
+                        pathExt = ".com;.exe;.bat;.cmd;.vbs;.vbe;.js;.jse;.wsf;.wsh";
+                    }
+
+                    string[] pathExtSegments = pathExt.Split(new string[] { ";" }, StringSplitOptions.RemoveEmptyEntries);
+
+                    // if command already has an extension.
+                    if (pathExtSegments.Any(ext => command.EndsWith(ext, StringComparison.OrdinalIgnoreCase)))
+                    {
+                        matches = Directory.GetFiles(pathSegment, command);
+                        if (matches != null && matches.Length > 0)
+                        {
+                            return matches.First();
+                        }
+                    }
+                    else
+                    {
+                        string searchPattern;
+                        searchPattern = StringUtil.Format($"{command}.*");
+                        matches = Directory.GetFiles(pathSegment, searchPattern);
+                        if (matches != null && matches.Length > 0)
+                        {
+                            // add extension.
+                            for (int i = 0; i < pathExtSegments.Length; i++)
+                            {
+                                string fullPath = Path.Combine(pathSegment, StringUtil.Format($"{command}{pathExtSegments[i]}"));
+                                if (matches.Any(p => p.Equals(fullPath, StringComparison.OrdinalIgnoreCase)))
+                                {
+                                    return fullPath;
+                                }
+                            }
+                        }
+                    }
+#else
+                    matches = Directory.GetFiles(pathSegment, command);
```

Comment:
Does .Net Core return matches based on the Windows short name? We might want to filter those out? Or maybe just add a comment explaining that it could happen? Or just leave it as is.